test/plots/barPlotTests.ts

```diff
@@ -839,138 +892,192 @@ describe("Plots", () => {
-        let texts = svg.selectAll("text")[0].map((n: any) => d3.select(n).text());
-        assert.lengthOf(texts, 0, "by default, no texts are drawn");
+        barPlot.renderTo(svg);
+        let texts = barPlot.content().selectAll("text");
+        assert.strictEqual(texts.size(), 0, "by default, no texts are drawn");
         svg.remove();
       });
 
       it("bar labels render properly", () => {
-        plot.renderTo(svg);
-        plot.labelsEnabled(true);
-        let texts = svg.selectAll("text")[0].map((n: any) => d3.select(n).text());
-        assert.lengthOf(texts, 2, "both texts drawn");
-        assert.strictEqual(texts[0], "640", "first label is 640");
-        assert.strictEqual(texts[1], "12345", "first label is 12345");
+        barPlot.renderTo(svg);
+        barPlot.labelsEnabled(true);
+        let texts = barPlot.content().selectAll("text");
+        let data = dataset.data();
+        assert.strictEqual(texts.size(), data.length, "one label drawn per datum");
+        texts.each(function(d, i) {
+          assert.strictEqual(d3.select(this).text(), data[i].y.toString(), `by default, label text is the bar's value (index ${i})`);
+        });
         svg.remove();
       });
 
       it("bar labels hide if bars too skinny", () => {
-        plot.labelsEnabled(true);
-        plot.renderTo(svg);
-        plot.labelFormatter((n: number) => n.toString() + (n === 12345 ? "looong" : ""));
-        let texts = svg.selectAll("text")[0].map((n: any) => d3.select(n).text());
-        assert.lengthOf(texts, 0, "no text drawn");
+        svg.attr("width", 100);
+        barPlot.labelsEnabled(true);
+        barPlot.renderTo(svg);
+        let texts = barPlot.content().selectAll("text");
+        assert.strictEqual(texts.size(), 0, "no labels drawn");
         svg.remove();
       });
 
       it("formatters are used properly", () => {
-        plot.labelsEnabled(true);
-        plot.labelFormatter((n: number) => n.toString() + "%");
-        plot.renderTo(svg);
-        let texts = svg.selectAll("text")[0].map((n: any) => d3.select(n).text());
-        assert.lengthOf(texts, 2, "both texts drawn");
-        assert.strictEqual(texts[0], "640%", "first label is 640%");
-        assert.strictEqual(texts[1], "12345%", "first label is 12345%");
+        barPlot.labelsEnabled(true);
+        let formatter = (n: number) => n.toString() + "%";
+        barPlot.labelFormatter(formatter);
+        barPlot.renderTo(svg);
+        let texts = barPlot.content().selectAll("text");
+        assert.strictEqual(texts.size(), dataset.data().length, "one label drawn per datum");
+        let expectedTexts = dataset.data().map((d) => formatter(d.y));
+        texts.each(function(d, i) {
+          assert.strictEqual(d3.select(this).text(), expectedTexts[i], `formatter is applied to the displayed value (index ${i})`);
+        });
+        svg.remove();
+      });
+
+      it("bar labels are shown inside or outside the bar as appropriate", () => {
```

Comment:
Can we ake this into a smooth sentence if possible?

`it("shows labels inside or outside the bar as appropriate, ...")`